docs/arcaflow/workflows/expressions.md

```diff
@@ -28,31 +70,256 @@ $.foo.bar
+
+All cases:
+| Left    | Right   | `&&`    |
+| ------- | ------- | ------- |
+| `true`  | `true`  | `true`  |
+| `true`  | `false` | `false` |
+| `false` | `true`  | `false` |
+| `false` | `false` | `false` |
+
+### Logical OR
+
+The `&&` operator returns `true` if either or both of the left and right sides are `true`, and `false` otherwise. It is a non-exclusive or operator. This operator requires boolean input.
+Note: There is no short-circuiting as it's currently implemented. Both the left and right are evaluated before the comparison takes place.
+
+All cases:
+| Left    | Right   | `\|\|`  |
+| ------- | ------- | ------- |
+| `true`  | `true`  | `true`  |
+| `true`  | `false` | `true`  |
+| `false` | `true`  | `true`  |
+| `false` | `false` | `false` |
+
+## Unary Operations
+
+These are operations that have one operator before the expressions.
+
+| Operator | Description        |
+| ---------|--------------------|
+| -        | [Negation](#Negation)           |
+| !        | [Logical complement](#logical-complement) |
+
+### Negation
+
+The negation operator negates the value from the input expression.
+The required format is a dash `-` before the expression.
+
+If the input is not numeric, there will be a type error.
+
+Examples with integer literals: `-5`, `- 5`
+Example with a float literal: `-50.0`
+Example with a reference: `-$.foo`
+
+### Logical complement
+
+The logical complement unary operator logically inverts the boolean input.
+The required format is an exclamation point `!` before the expression.
+
+If the input expression is not boolean, there will be a type error.
+
+Example with a boolean literal: `!true`
+Example with a reference: `!$.foo`
+
+## Parentheses
+
+Parentheses are used to force precedence in the expression. They do not do anything implicitly (for example, there is no implied multiplication)
+
+For example, the expression `5 + 5 * 5` evaluates the `5 * 5` before the `+`, resulting in `5 + 25`, and finally `30`.
+If you want the 5 + 5 to be run first, you must use parentheses. That gives you the expression `(5 + 5) * 5`, resulting in `10 * 5`, and finally `50`
+
+## Order of Operations
+
+The order of operations is designed to match most programming languages, and the rules of math.
+
+Order (highest to lowest):
+ - [`- ` negation](#negation)
+ - [`()` parentheses](#parentheses)
+ - [`^ ` exponent](#exponentiation)
+ - [`* ` multiplication](#multiplication) and [division `/`](#division)
+ - [`+ ` addition/concatenation](#additionconcatenation) and [subtraction `-`](#subtraction)
+ - binary equality and inequality (all equal)
+   - [`==` equals](#equals)
+   - [`!=` not equals](#not-equals)
+   - [`> ` greater than](#greater-than)
+   - [`< ` less than](#less-than)
+   - [`>=` greater than or equal to](#greater-than-or-equal-to)
+   - [`<=` less than or equal to](#less-than-or-equal-to)
+ - [`! ` logical complement](#logical-complement)
```

Comment:
Fascinating:  I think I would have expected the complement operation to be higher precedence than the relational operators...but I think it turns out not to matter -- that is, I think that all three of these produce the same results:
```
(!$.a) == false
!($.a == false)
$.a != false
```
And, so, since we don't allow `!` to be applied to anything other than booleans, and since `==` and `!=` are the only relational operators which accept booleans, I guess the precedence doesn't matter.